### DISCUSSION TITLE
ngtcp2 with picotls support

### DISCUSSION THREAD

**Discussion by liudongmiao** [curl/curl#13572] [Q&A] [answered]
For openssl, there is two port:
- quictls, openssl with quic patch
- picotls, quic with several tls libraries include openssl

currently, curl with ngtcp2 support all tls libraries supported by ngtcp2 except picotls.

Is there any chance for curl to support ngtcp2 with picotls?

**Comment by icing** [comment, marked as answer]
Currently, the ngtcp2 TLS library has to be the same as for the rest of curl. So, to use picotls for H3, curl would need to support it also for all other TLS connections.

There has been some discussion about decoupling this, but it makes things like Session sharing and various libcurl callbacks complicated. We decided not to go that route.

**Comment by bagder** [comment]
To extend on @icing's answer: it is *possible*, but someone would have to write the code.
